### PR TITLE
Bookstore redux

### PR DESCRIPTION
**Pull request by getssh**
- Added react-redux and redux-toolkit
- Added booksSlice
- Added catagoriesSlice
- Added store
- Used recommended file structure
- Checked linter

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    addBook: (state, action) => {
+      state.books.push({
+        bookId: uuid(),
+        name: action.payload.name,
+        author: action.payload.author,
+      });
+    },
+    removeBook: (state, action) => {
+      state.books.filter((book) => book.id !== action.payload.id);
+    },
```
Comment (amon-cofie) [inline src/redux/books/booksSlice.js]: You have done a very good job putting all the reducers into the redux store and I must say, you did it very well 👍 However, please note that the filter method returns a new array and hence you have to assign that new array to the state. Along the line you will learn about immutability and get to know a better to do this but for now, you can change line 18 to look something like this: 

```sh
   state.books = state.books.filter((book) => book.id !== action.payload.id);
```

**Review by amon-cofie — CHANGES_REQUESTED** [review]
Hi @getssh , 👋 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

🚀 You have no linter errors
🚀 You have a descriptive pull request summary
🚀 You have descriptive commit messages

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (rloterh) [reply src/redux/books/booksSlice.js]: resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Getayawkal,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 👍
- Good job implementing recommended changes!

Cheers..Happy coding! 💻  🍷  🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**